xonsh/completer.py

```diff
@@ -312,19 +322,28 @@ def bash_complete(self, prefix, line, begidx, endidx):
         return rtn
 
     def _source_completions(self):
+        """Return list of source commands for Bash completion files.
+
+        Note that this silently ignores any file name in the completion list
+        that does not exist.
+        """
+        _ = self
         srcs = []
         for f in builtins.__xonsh_env__.get('BASH_COMPLETIONS'):
             if os.path.isfile(f):
-                # We need to "Unixify" Windows paths for Bash to understand
                 if ON_WINDOWS:
-                    f = RE_WIN_DRIVE.sub(lambda m: '/{0}/'.format(m.group(1).lower()), f).replace('\\', '/')
+                    # "Unixify" Windows paths for Bash to understand them.
+                    f = RE_WIN_DRIVE.sub(
+                        lambda m: '/{0}/'.format(m.group(1).lower()),
+                        f).replace('\\', '/')
                 srcs.append('source ' + f)
         return srcs
 
     def _load_bash_complete_funcs(self):
+        """Attempt to load any Bash completion functions we can find."""
         self.bash_complete_funcs = bcf = {}
         inp = self._source_completions()
-        if len(inp) == 0:
+        if not inp:  # no completion files could be found
```

Comment:
Again, sorry, but this is Google style and even if I weren't a recent Google employee would agree with that style rule.

The predicate is whether or not `inp` has been set to a value that isn't empty (e.g., `[]`) or undefined (e.g., None). If `inp` being an empty sequence versus `None` was important then that would be tested for. The "len(...) == 0" idiom makes sense for other language but not Python.